bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/ErrorValue.java

```diff
@@ -73,21 +78,54 @@ public ErrorValue(BType type, BString message, ErrorValue cause, Object details)
     }
 
     @Override
-    public String stringValue() {
+    public String stringValue(BLink parent) {
+        CycleUtils.Node linkParent = new CycleUtils.Node(this, parent);
         if (isEmptyDetail()) {
-            return "error " + message.getValue();
+            return "error" + getModuleName() + "(" + ((StringValue) message).informalStringValue(linkParent) + ")";
         }
-        return "error " + message.getValue() + " " + getCauseToString() +
-                org.ballerinalang.jvm.values.utils.StringUtils.getStringValue(details);
+        return "error" + getModuleName() + "(" + ((StringValue) message).informalStringValue(linkParent) +
+                getCauseToString(linkParent) + getDetailsToString(linkParent) + ")";
     }
 
-    private String getCauseToString() {
+    private String getCauseToString(BLink parent) {
         if (cause != null) {
-            return org.ballerinalang.jvm.values.utils.StringUtils.getStringValue(cause) + " ";
+            return ", " + cause.informalStringValue(parent);
         }
         return "";
     }
 
+    private String getDetailsToString(BLink parent) {
+        StringJoiner sj = new StringJoiner(", ");
+        for (Object key : ((MapValue) details).getKeys()) {
+            Object value = ((MapValue) details).get(key);
+            if (value == null) {
+                sj.add(key + "=null");
+            } else {
+                BType type = TypeChecker.getType(value);
```

Comment:
wondering whether we can reuse and share between error and mapvalueImpl stringValue methods @mohanvive wdyt?

Reply:
from line 104 to 120

Reply:
BTW, there should be space between equal and null right? Isn't it ?

Reply:
I think, there is a delimeter difference between them.

Reply:
Should there be a space specifically for `null`? 
In James's example here https://github.com/ballerina-platform/ballerina-spec/issues/499#issuecomment-690819039 , there's no space between the key and the value (`accountID=123` and `detail="Failed to get account balance"`)

Reply:
Ah, Ok. Cool.

Reply:
Yes. There's a delimiter difference. map uses `{"key1": "value1", "key1": "value2"}` format. here it should be `key1="value1", key2="value2"`